patches/tModLoader/Terraria/ModLoader/Core/ModOrganizer.cs

```diff
@@ -319,5 +322,70 @@ internal static class ModOrganizer
 				return new HashSet<string>();
 			}
 		}
+
+		internal static string GetActiveTmodInRepo(string repo) {
+			Version tmodVersion = new Version(BuildInfo.tMLVersion.Major, BuildInfo.tMLVersion.Minor);
+			string[] tmods = Directory.GetFiles(repo, "*.tmod", SearchOption.AllDirectories);
+			if (tmods.Length == 1)
+				return tmods[0];
+
+			string val = null;
+			Version currVersion = null;
+			foreach (string fileName in tmods) {
+				var regex = new Regex(@"(\d\d\d\d[.]\d\d)");
```

Comment:
it looks like you're putting the tMod version in the mod filename? This might be a good idea, I think my original expectation was that we'd put them in subfolders.

I'd prefer if we actually read the mod file header, but there isn't actually a `TmodFile` method to just read to the tML version. Though it's the first string after the magic `TMOD` bytes so it's not expensive to get.

Reply:
the 'filename' string is still the full file name, ie includes parent folders. this would work with both subfolders or if we wanted to rename the tmods. 
Let's stick with subfolders in case other files crop up in some mods.